engine/src/main/java/org/destinationsol/game/SolGame.java

```diff
@@ -208,10 +236,72 @@ private void createPlayer(String shipName) {
         objectManager.resetDelays();
     }
 
+    /**
+     * Creates and spawns the players mercenaries from their JSON file.
+     */
+    private void createMercs() {
+        String fileName = "mercenaries.json";
+
+        if (!SaveManager.resourceExists(fileName)) {
+            return;
+        }
+
+        String path = SaveManager.getRsrcPath(fileName);
+        BufferedReader bufferedReader;
+        try {
+            bufferedReader = new BufferedReader(new FileReader(path));
+            if (new File(path).length() == 0) {
+                bufferedReader.close();
+                return;
+            }
+        } catch (IOException e) {
+            return;
+        }
+
+        Gson gson = new Gson();
+        Type type = new TypeToken<ArrayList<HashMap<String, String>>>() {
+        }.getType();
+        ArrayList<HashMap<String, String>> mercs = gson.fromJson(bufferedReader, type);
+
+        MercItem mercItem;
+        for (HashMap<String, String> node : mercs) {
+            mercItem = new MercItem(
+                    new ShipConfig(hullConfigManager.getConfig(node.get("hull")), node.get("items"), Integer.parseInt(node.get("money")), -1f, null, itemManager));
+            MercenaryUtils.createMerc(this, hero, mercItem);
+        }
+
+    }
+
     public void onGameEnd() {
         saveShip();
+        saveWorld();
         objectManager.dispose();
     }
+    
+    /**
+     * Saves the world's seed so we can regenerate the world the same next time
```

Comment:
Rephrase.

`regenerate the world the same`
Makes you go `wut?`

`regenerate the same world`